### PR TITLE
fix(components/text-editor): content is only pasted once into the text editor

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
[AB#2451413](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2451413)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/997/storybook/)

Component Storybooks:

- _(no component storybooks affected in this pr)_ 

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/997/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/997/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/997/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/997/playground/)

File: libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.spec.ts [hunk 1]
```diff
@@ -1333,6 +1334,10 @@ describe('Text editor', () => {
     }));
 
     it('should set the style of the iframe body to the provided style state', fakeAsync(() => {
+      // We normally load the iframe in the `beforeEach`. However, we need to reset things here so that the initial styles can be applied first.
```
Comment (Blackbaud-SteveBrush) [inline libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.spec.ts:1337]: This is where the `setupTest` function would be useful (future consideration). Thanks for the comment.

File: libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.ts [hunk 2]
```diff
@@ -376,66 +363,68 @@ export class SkyTextEditorComponent implements AfterViewInit, OnDestroy {
...
+          this.editorFocusStream.next();
+        });
 
-    this.#adapterService.setEditorInnerHtml(this.#_value);
+      this.#editorService
+        .blurListener()
+        .pipe(takeUntil(this.#ngUnsubscribe))
+        .subscribe(() => {
+          // Angular doesn't run change detection for changes originating inside an iframe,
+          // so we have to run markForCheck() inside the NgZone to force change propagation to consuming components.
```
Comment (Blackbaud-SteveBrush) [inline libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.ts]: Should `markForCheck` in the comment be `#_onTouched`?

Comment (Blackbaud-TrevorBurch) [reply libs/components/text-editor/src/lib/modules/text-editor/text-editor.component.ts]: Updated - this was an existing comment - but does seem that it needed updated. Done

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**